### PR TITLE
HOTFIX set Background brick did not work anymore

### PR DESCRIPTION
**Pull request by p4p4**
1. commit introduced in PR #1923 broke the SetBackgroundBrick
    -> fixed
2. commit 6c25a4e448e8f0dc165a05b688e6271d87d5784f replaced wrong strings on brick: ´
      should be `Switch to Background [...] and wait`, not `Switch to look [...] Background`

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetBackgroundBrick.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public SetBackgroundBrick() {
 	}
 
 	@Override
-	protected Sprite getSprite() {
+	protected Sprite getSpriteToChange() {
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetBackgroundBrick.java]: Please revert this change. It's not important for the current release. I'll explain to you what I meant at https://github.com/Catrobat/Catroid/pull/1923#discussion_r83840129 later. :wink:

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 2]
```diff
@@ -182,18 +182,23 @@ public Brick clone() {
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
-		sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		if (this instanceof SetBackgroundBrick) {
+			Sprite backgroundSprite = getSpriteToChange();
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(backgroundSprite, look, wait));
+		} else {
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		}
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: Is this change really necessary?
The sprite should always be the correct sprite, either a normal sprite or a background.

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 3]
```diff
@@ -182,7 +182,8 @@ public Brick clone() {
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
-		sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		sequence.addAction(sprite.getActionFactory().createSetLookAction(getSprite(), look, wait));
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: Please revert

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java (hunk 2 above)
Comment (p4p4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: ok then back to you initial question:
 - yes I think it should be necessary to do this:

if this is a `SetBackgroundBrick`, then use `getSprite()`, which [is overridden](https://github.com/Catrobat/Catroid/blob/b5a9fc8323373031b4d561cba3d69de77d094319/catroid/src/main/java/org/catrobat/catroid/content/bricks/SetBackgroundBrick.java#L36) for `SetBackgroundBrick` and always returns the background sprite.
in all other cases, the passed `sprite` is used.

Apparantly it seems to be time to stop working for today... :smiley:

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 4]
```diff
@@ -182,7 +182,12 @@ public Brick clone() {
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
-		sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		if (this instanceof SetBackgroundBrick) {
+			Sprite backgroundSprite = getSprite();
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(backgroundSprite, look, wait));
+		} else {
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		}
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: You could also override the method in the SetBackgroundBrick :wink:

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java (hunk 2 above)
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: We definitely have to add some tests for the BackgroundBricks

Comment (p4p4) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: you are right, for the action it is necessary to use the "SceneToPlay", whereas the Spinners in the IDE need the "CurrentScene".

If anybody has time for writing a test she is welcome to do so.
My previous hotfixes were only on a voluntary basis to help out for the next release :grinning:

**Comment by robertpainsi** [comment]
@p4p4 thanks for fixing it!
